console/console-init/ui/src/components/DropdownWithToggle/DropdownWithToggle.tsx

```diff
@@ -71,6 +70,20 @@ export const DropdownWithToggle: React.FC<IDropdownWithToggleProps &
     element && element.focus();
   };
 
+  /*
+   *  If there is only one option use it as default
+   */
+  useEffect(() => {
+    if (!value || value.trim() === "") {
+      if (dropdownItems && dropdownItems.length === 1) {
+        const item = dropdownItems[0];
+        if (item && item.value && onSelectItem) {
+          onSelectItem(item.value);
+        }
+      }
+    }
+  }, [dropdownItems]);
```

Comment:
it would be better to add value dependency to avoid console warning and safer side  as your concern about re run the code which will handle by icondition if (!value || value.trim() === "").